ui/guest/src/contentController.ts

```diff
@@ -385,6 +385,16 @@ export function insertItem(modelId: string, fieldId: string, index: number | str
     }
   });
 
+  const currentItems = [...(models[modelId][fieldId] ?? [])];
+  const newItems = index ? currentItems.splice(index, 0, instance) : [...currentItems, instance];
+  models$.next({
+    ...models,
+    [modelId]: {
+      ...models[modelId],
+      [fieldId]: newItems
+    }
+  });
```

Comment:
Create a nested fields/indexes example (e.g. repeat_o.itemSelector_o, 0.1) and see if this works. My guess if we need to accommodate the dotted notation by using the utility functions we have.